### PR TITLE
Show password config on validation failure

### PR DESCRIPTION
**Pull request by bdjelaili**
closes gravitee-io/issues#5128

### CODE REVIEW

File: gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/main/java/io/gravitee/am/gateway/handler/common/utils/ConstantKeys.java [hunk 1]
```diff
@@ -41,6 +41,7 @@
     String ERROR_DESCRIPTION_PARAM_KEY = "error_description";
     String SUCCESS_PARAM_KEY = "success";
     String WARNING_PARAM_KEY = "warning";
+    String PASSWORD_CONFIG_PARAM_KEY = "password_config";
```
Comment (tcompiegne) [inline gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/main/java/io/gravitee/am/gateway/handler/common/utils/ConstantKeys.java:44]: I prefer camelCase when dealing with Thymeleaf template to keep a consistency with the other keys.

File: gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/java/io/gravitee/am/gateway/handler/root/resources/endpoint/user/password/ResetPasswordEndpoint.java [hunk 2]
```diff
@@ -34,45 +33,46 @@
...
-        final String error = request.getParam(ConstantKeys.ERROR_PARAM_KEY);
-        final String errorDescription = request.getParam(ConstantKeys.ERROR_DESCRIPTION_PARAM_KEY);
-        final String success = request.getParam(ConstantKeys.SUCCESS_PARAM_KEY);
-        final String warning = request.getParam(ConstantKeys.WARNING_PARAM_KEY);
-        final String token = request.getParam(TOKEN_PARAM_KEY);
-        // add query params to context
+        HttpServerRequest request = routingContext.request();
+        // retrieve client (if exists)
+        Client client = routingContext.get(ConstantKeys.CLIENT_CONTEXT_KEY);
+        routingContext.put(ConstantKeys.PASSWORD_CONFIG_PARAM_KEY, PasswordSettings.getInstance(client, domain));
```
Comment (tcompiegne) [inline gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/java/io/gravitee/am/gateway/handler/root/resources/endpoint/user/password/ResetPasswordEndpoint.java:57]: Do not inject the Optional here but rather the value, this way we keep coherency with the others attributes.

File: gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/java/io/gravitee/am/gateway/handler/root/resources/endpoint/user/register/RegisterEndpoint.java [hunk 3]
```diff
@@ -40,30 +42,33 @@
...
-        final String errorDescription = request.getParam(ConstantKeys.ERROR_DESCRIPTION_PARAM_KEY);
-        final String success = request.getParam(ConstantKeys.SUCCESS_PARAM_KEY);
-        final String warning = request.getParam(ConstantKeys.WARNING_PARAM_KEY);
-        final Client client = routingContext.get(ConstantKeys.CLIENT_CONTEXT_KEY);
-        final String clientId = request.getParam(Parameters.CLIENT_ID);
+        HttpServerRequest request = routingContext.request();
+        copyValue(request, routingContext, ConstantKeys.SUCCESS_PARAM_KEY);
+        copyValue(request, routingContext, ConstantKeys.WARNING_PARAM_KEY);
+        Client client = routingContext.get(ConstantKeys.CLIENT_CONTEXT_KEY);
+        routingContext.put(ConstantKeys.PASSWORD_CONFIG_PARAM_KEY, PasswordSettings.getInstance(client, domain));
```
Comment (tcompiegne) [inline gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/java/io/gravitee/am/gateway/handler/root/resources/endpoint/user/register/RegisterEndpoint.java:63]: Do not inject the Optional here but rather the value, this way we keep coherency with the others attributes.

**Review by tcompiegne — CHANGES_REQUESTED** [review]

File: gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/main/java/io/gravitee/am/gateway/handler/common/utils/ConstantKeys.java (hunk 1 above)
Comment (bdjelaili) [reply gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/main/java/io/gravitee/am/gateway/handler/common/utils/ConstantKeys.java]: ok

File: gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/java/io/gravitee/am/gateway/handler/root/resources/endpoint/user/password/ResetPasswordEndpoint.java (hunk 2 above)
Comment (bdjelaili) [reply gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/java/io/gravitee/am/gateway/handler/root/resources/endpoint/user/password/ResetPasswordEndpoint.java]: I forget that PasswordSettings.getInstance() return an optional, thank's

File: gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/java/io/gravitee/am/gateway/handler/root/resources/endpoint/user/register/RegisterEndpoint.java (hunk 3 above)
Comment (bdjelaili) [reply gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/java/io/gravitee/am/gateway/handler/root/resources/endpoint/user/register/RegisterEndpoint.java]: ok

**Pull request closed without merging**